src/plugins/data/common/search/types.ts

```diff
@@ -17,6 +17,22 @@
  * under the License.
  */
 
+import { Observable } from 'rxjs';
+import { IEsSearchRequest, IEsSearchResponse, ISearchOptions } from '../../common/search';
+
+export type ISearch = (
+  request: IKibanaSearchRequest,
+  options?: ISearchOptions
+) => Observable<IKibanaSearchResponse>;
+
+export type ISearchGeneric = <
+  SearchStrategyRequest extends IEsSearchRequest = IEsSearchRequest,
+  SearchStrategyResponse extends IEsSearchResponse = IEsSearchResponse
+>(
+  request: SearchStrategyRequest,
+  options?: ISearchOptions
+) => Observable<SearchStrategyResponse>;
+
```

Comment:
For the purposes of this PR, these were the only types from `public/search/types` which needed to be relocated to `common`.